cmd/coder/sync.go

```diff
@@ -79,8 +81,10 @@ func (cmd *syncCmd) Run(fl *pflag.FlagSet) {
 		err = s.Run()
 	}
 
-	if err == NoRsync {
+	if fmt.Sprintf("%v", err) == fmt.Sprintf("%v", IncompatRsync) {
 		flog.Fatal("no compatible rsync present on remote machine")
+	} else if fmt.Sprintf("%v", err) == fmt.Sprintf("%v", StreamErrRsync) {
+		flog.Fatal("error in rsync protocol datastream (no installed remote rsync?)")
```

Comment:
This isn't great.

We should be checking the exit code [here](https://github.com/cdr/coder-cli/blob/master/internal/sync/sync.go#L67), and wrap the error with an informative string based on the exit code.

See https://stackoverflow.com/a/55055100 for how we can get the exit code from a command error.

Reply:
Yeah, I figured this wasn't be best way to do it :/ I'll update this ASAP.